docs/source/l1_to_l2b_e2e.rst

```diff
@@ -0,0 +1,232 @@
+
+    - Detector Noise Maps: quantifies different noise sources from the sensor.
+    - Non-Linearity: Corrects for any non-linearities in the sensor's response.
+    - Darks: generated from Detector Noise Maps to account for dark current in the sensor.
+    - Flat Field: Used to correct for uneven sensitivity across the detector.
+    - Bad Pixel Maps: Identifies pixels that are defective or not functioning properly.
+
+
+Setup
+~~~~~~
+
+Before running the test, ensure you have the necessary datasets and calibration files. You will need:
+
+1. **L1 Data** - Raw Level 1 data files.
+2. **Cals** - Calibration files required for processing as outlined above. 
+
+The test script is written to process the L1 data, apply the calibration files, and produce the output in the L2b format.
+
+
+.. code-block:: python
+
+    import argparse
+    import os
+    import pytest
+    import numpy as np
+    import astropy.time as time
+    import astropy.io.fits as fits
+    import corgidrp
+    import corgidrp.data as data
+    import corgidrp.mocks as mocks
+    import corgidrp.walker as walker
+    import corgidrp.caldb as caldb
+    import corgidrp.detector as detector
+
+
+Before processing the L1 files, the TVAC FITS headers must be standardized using the ``fix_headers_for_tvac`` function:
+
+Helper function to standardize TVAC FITS headers for pipeline compatibility:
+
+.. code-block:: python
+
+    def fix_headers_for_tvac(list_of_fits):
+    print("Fixing TVAC headers")
+    for file in list_of_fits:
+            fits_file = fits.open(file)
+            prihdr = fits_file[0].header
+            exthdr = fits_file[1].header
+            prihdr['VISTYPE'] = "TDEMO"
+            prihdr['OBSNUM'] = prihdr['OBSID']
+            exthdr['EMGAIN_C'] = exthdr['CMDGAIN']
+            exthdr['EMGAIN_A'] = -1
+            exthdr['DATALVL'] = exthdr['DATA_LEVEL']
+            exthdr['KGAINPAR'] = exthdr.get('KGAIN', 8.7)
+            prihdr["OBSNAME"] = prihdr['OBSTYPE']
+            prihdr['PHTCNT'] = False
+            exthdr['ISPC'] = False
+            fits_file.writeto(file, overwrite=True)
+
+This function modifies TVAC FITS headers to match the structure expected by the pipeline.
+
+Set up directory paths for input and output data:
+
+.. note::
+    The following calibration setup is specifically for the example TVAC test data. For real data processing, users will either:
+    
+    1. Have downloaded pre-processed calibration FITS files that just need to be added to the caldb
+    2. Have processed the calibrations themselves locally, in which case the calibration files should already exist in their caldb (located in ``~/.corgidrp/``)
+    
+    The code below demonstrates how we adapt the TVAC test data to fit the official corgidrp calibration format. This is not the typical workflow for real data processing.
```

Comment:
Add another sentence saying that we return to the intended workflow at the start of the "Pipeline Execution" section.